src/csharp/akka_validation/Network.cs

```diff
@@ -166,11 +166,12 @@ protected override void OnReceive(object message)
             }
             else if (message is Subscribe subscribe)
             {
-                if (!_subscribers.ContainsKey(subscribe.Topic))
+                var topic = subscribe.Topic;
+                if (!_subscribers.ContainsKey(topic))
                 {
-                    _subscribers = _subscribers.Add(subscribe.Topic, ImmutableList<IActorRef>.Empty);
+                    _subscribers = _subscribers.Add(topic, ImmutableList<IActorRef>.Empty);
                 }
-                _subscribers.SetItem(subscribe.Topic, _subscribers[subscribe.Topic].Add(subscribe.Subscriber));
+                _subscribers = _subscribers.SetItem(topic, _subscribers[topic].Add(subscribe.Subscriber));
```

Comment:
Subscribers was immutable, we thus actually need to update the variable, otherwise no subscribers will be added.